jinahub/rankers/SimpleRanker/simpleranker.py

```diff
@@ -1,69 +1,111 @@
+            want the ranker to work on - these are the "query" documents, for which
+            we wish to create aggregated matches.
+        :param matches_path: The path (starting from the "query" document) where the
+            matched chunks can be found. The default here is "cm", which means that
+            we expecte the matched chunks to be the matches of the chunks of our query
+            document. This path should end with "m"
         """
         super().__init__(*args, **kwargs)
+
+        if ranking not in ['min', 'max', 'mean_min', 'mean_max']:
+            raise ValueError(
+                'ranking should be one of "min", "max", "mean_min", "mean_max"',
+                f' got "{ranking}"',
+            )
+
+        if matches_path[-1] != "m":
+            raise ValueError(f'matches_path should end with "m", got {matches_path}')
+
         self.metric = metric
-        assert ranking in ['min', 'max', 'mean_min', 'mean_max']
         self.ranking = ranking
         self.traversal_paths = traversal_paths
+        self.matches_path = matches_path
 
     @requests(on='/search')
-    def rank(self, docs: DocumentArray, parameters: Dict, *args, **kwargs):
-        traversal_paths = parameters.get('traversal_paths', self.traversal_paths)
+    def rank(
+        self, docs: Optional[DocumentArray] = None, parameters: Dict = {}, **kwargs
+    ):
+        """Aggregate the score of matched chunks to the score of their parent document.
+
+        The matches of query documents that are passed in `docs` are replaced (if they
+        exist) by documents based on the aggregated score of the parent documents of
+        matched chunks - that is, by documents containing only parent id of matched
+        chunks, and the aggregated score corresponding to that id.
 
+        :param docs: The documents for which to create aggregated matches (specifically,
+            the aggregated matches will be created for documents that are on the
+            traversal paths of documents passed in this argument).
+        :param parameters: Extra parameters that can be used to override the parameters
+            set at creation of the Executor. Valid values are `traversal_paths`
+        """
+        if docs is None:
+            return
+
+        traversal_paths = parameters.get('traversal_paths', self.traversal_paths)
         for doc in docs.traverse_flat(traversal_paths):
-            matches_of_chunks = []
-            for chunk in doc.chunks:
-                matches_of_chunks.extend(chunk.matches)
-            groups = groupby(
-                sorted(matches_of_chunks, key=lambda d: d.parent_id),
-                lambda d: d.parent_id,
-            )
-            for key, group in groups:
-                chunk_match_list = list(group)
+            parents_scores = defaultdict(list)
+            for m in DocumentArray([doc]).traverse_flat([self.matches_path]):
```

Comment:
```suggestion
            for m in DocumentArray([doc]).traverse_flat(['m']):
```

Reply:
@nan-wang If we want to remove this argument, shouldn't we leave it at `'cm'` ? This was the original default case, and is used like this in the multrires-lyrics-search example.

Reply:
Yes, you are right.